tasks/interview-corejs.md

```diff
@@ -1,51 +1,86 @@
-*  five(plus(seven(minus(three())))) -> 9. How to implement?
-*  add(5)(9)(-4)(1) -> 11. How to implement?
-* `periodOutput(period)` method should output in the console once per every period how much time left from the function first call. 
+## JS Questions:
+
+- ### Objects Built-in methods.
+
+  - Know how to use built-in methods
+  - Know static Object methods
+  - Property flags & descriptors (student is able to set property via Object. defineProperty)
+
+- ### Object as Hash.
+
+  - Be able to loop through Object keys
+  - Know how to create iterable objects, Symbol.iterator usage
+
+- ### Arrays Built-in methods
+
+  - Know how to copy array
+  - Know how to copy array part
+  - Know how to modify array
+
+- ### Arrays Iterating, Sorting, Filtering
+
+  - Know how to sort Array
+  - Be able to custom sorting for Array
+  - Be able to filter Array elements
+  - Know several method how to iterate Array elements
+
+- ### Functional Scope
+
+  - Know global scope and functional scope
+  - Know variables visibility areas
+  - Understand nested scopes and able work with them
+
+- ### Functions Parameters / Arguments
+
+  - Know how to define Function parameters
+  - Know difference between parameters passing by value and by reference
+  - Know how to handle dynamic amount of Function parameters
+
+- ### ECMAScript Intermediate
+  - Function default parameters
+  - ECMAscript modules
+  - Know how to use spread operator for Function arguments
+  - Be able to compare [arguments] and spread operator
+  - Spread operator for Array
+  - Understand and able to use spread operator for Array concatenation
+    Destructuring assignment
+  - Be able to discover destructuring assignmentconcept
+  - Understand variables and Function arguments destructuring assignment
+  - String templates
+  - Know how [for..of] loop works
+
+## Coding tasks:
+
+- `Function.prototype.bind` implement polyfill
+- `Object.create` implement polyfill
+- `Array.flat` implement polyfill
+- 'hello world'.repeating(3) -> 'hello world hello world hello world'. How to implement?
+- myFunc('!', 4, -10, 34, 0) -> '4!-10!34!0`. How to implement?
+- five(plus(seven(minus(three())))) -> 9. How to implement?
+- add(5)(9)(-4)(1) -> 11. How to implement?
+- `periodOutput(period)` method should output in the console once per every period how much time left from the function first call.
   Example:
   `periodOutput(100) -> 100(after 100 ms), 200(after 100 ms), 300(after 100 ms), ...`
-* `extendedPeriodOutput(period)` method should output in the console once per period how much time left from the function first call and increase the period. Example: `// extendedPeriodOutput(100) -> 100(after 100 ms), 200(after 200 ms), 300(after 300 ms)`
-
+- `extendedPeriodOutput(period)` method should output in the console once per period how much time left from the function first call and increase the period. Example: `// extendedPeriodOutput(100) -> 100(after 100 ms), 200(after 200 ms), 300(after 300 ms)`
```

Comment:
... how mach time has passed since the first function call and then increase the period